src/tests/ftest/util/test_utils_pool.py

```diff
@@ -61,6 +72,41 @@ def __init__(self, context, dmg_command, cb_handler=None):
         self.dmg = dmg_command
         self.query_data = []
 
+    def get_params(self, test):
+        """Get values for all of the command params from the yaml file.
+
+        Autosize any size/scm_size/nvme_size parameter whose value ends in "%".
+
+        Args:
+            test (Test): avocado Test object
+        """
+        super().get_params(test)
+
+        # Autosize any size/scm_size/nvme_size parameters
+        if ((self.size.value is not None and str(self.size.value).endswith("%"))
+                or (self.scm_size.value is not None
+                    and str(self.scm_size.value).endswith("%"))
+                or (self.nvme_size.value is not None
+                    and str(self.nvme_size.value).endswith("%"))):
+            index = self.server_index.value
+            try:
+                params = test.server_managers[index].autosize_pool_params(
+                    self.size.value, self.scm_ratio.value, self.scm_size.value,
+                    self.nvme_size.value, self.quantity.value,
+                    self.min_targets.value)
+            except ServerFailed as error:
+                test.fail(
+                    "Failure autosizing pool parameters: {}".format(error))
+
+            # Update the pool parameters with any autosized values
+            for name in params:
+                test_pool_param = getattr(self, name)
+                test_pool_param.update(params[name], name)
+
+                # Cache the autosized value so we do not calculate it again
+                id = (name, self.namespace, test_pool_param._default)
```

Comment:
(pylint-redefined-builtin) Redefining built-in 'id'
(pylint-protected-access) Access to a protected member _default of a client class